### DISCUSSION TITLE
The team_id is being written as null after upgrading from Laravel 10 to 11

### DISCUSSION THREAD

**Discussion by davidyell** [spatie/laravel-permission#2677] [Q&A]
### Summary

I am currently working through the Laravel Shift bits to upgrade a project from Laravel 10 to 11. We make use of the permissions package, and have teams enabled.

However, we are encountering issues in the test suite when running our Feature tests, which use `$this->user->assignRole('admin');`. These tests are all producing a MySQL constraint exception because the `team_id` cannot be found.

```bash
  FAILED  Tests\Feature\Livewire\Tags\TagsIndexTest > tag index component renders wi…  QueryException   
  SQLSTATE[23000]: Integrity constraint violation: 19 NOT NULL constraint failed: model_has_roles.team_id (Connection: sqlite, SQL: insert into "model_has_roles" ("model_id", "model_type", "role_id", "team_id") values (5, App\Models\User, 2, ?))
```

### Things I've tried

I have chased the code for the teams through and found the helpers `setPermissionsTeamId()` and `getPermissionsTeamId()`, but adding these I still get an error.

I can `dd()` both the session `team_id` and also the `getPermissionsTeamId()` and they return the correct value.

I also tried flushing the permissions using `forgetCachedPermissions()` in the `PermissionRegistrar` class.

Any thoughts?

**Comment by parallels999** [comment]
Laravel 11 seems to be working, look: https://github.com/spatie/laravel-permission/actions/runs/9239793952/job/25419362560
Maybe is a problem on your environment
Upload a demo repo on a fresh instalation  showing the problem
[https://spatie.be/laravel-permission/creating-a-demo-app](https://spatie.be/docs/laravel-permission/v6/basic-usage/new-app#content-creating-a-demo-app)

**Comment by davidyell** [comment]
I seem to have solved the issue by having to pull the permissions registrar from the container, set the team id and put it back in the container during my test before calling `assignRole()`

```php
        $registrar = app(PermissionRegistrar::class);
        $registrar->setPermissionsTeamId($this->team->id);
        $this->swap(PermissionRegistrar::class, $registrar);
```

Not sure why though

**Reply by erikn69** [reply]
That's very strange, it shouldn't happen.
Maybe you need something else, like this
https://github.com/spatie/laravel-permission/blob/c99dca701d60e1a2d1f839c9bfd102b8098432fe/tests/TestCase.php#L96-L99

**Reply by davidyell** [reply]
Even more strangely, this morning, just using the helper `setPermissionsTeamId()` seems to be working as expected! I wonder if perhaps I had some bad container cache 🤔
